internal/internal_worker_cache.go

```diff
@@ -55,55 +62,66 @@ var workerRefcount atomic.Int32
 // between workers running within same process. This must be called before any worker is started. If not called, the
 // default size of 10K (might change in future) will be used.
 func SetStickyWorkflowCacheSize(cacheSize int) {
-	stickyCacheLock.Lock()
-	defer stickyCacheLock.Unlock()
-	if workflowCache != nil {
+	sharedWorkerCacheLock.Lock()
+	defer sharedWorkerCacheLock.Unlock()
+	if sharedWorkerCachePtr.workflowCache != nil {
 		panic("cache already created, please set cache size before worker starts.")
 	}
-	stickyCacheSize = cacheSize
+	desiredWorkflowCacheSize = cacheSize
 }
 
-// Returns a pointer to the workerCache, and increases the outstanding refcount by one. Instances of
-// workerCache decrement the refcounter as a hook to runtime.SetFinalizer
-func getWorkerCache() *workerCache {
-	rcount := workerRefcount.Load()
-	if rcount == 0 {
-		stickyCacheLock.Lock()
-		defer stickyCacheLock.Unlock()
-		newcache := cache.New(stickyCacheSize, &cache.Options{
+// Creates a new workerCache, and increases workerRefcount by one. Instances of workerCache decrement the refcounter as
+// a hook to runtime.SetFinalizer (ie: When they are freed by the GC). When there are no reachable instances of
+// workerCache, shared caches will be cleared
+func newWorkerCache() *workerCache {
+	return _newWorkerCache(sharedWorkerCachePtr, &sharedWorkerCacheLock)
+}
+
+// This private version allows us to test functionality without affecting the global shared cache
```

Comment:
I would call the above method NewWorkerCache and this one newWorkerCache as Go uses capitalization of the first letter to indicate private/public